bin/metababel

```diff
@@ -87,30 +99,51 @@ end
 # Indeeed the <%= body %> will be indented, but we don't don't want it,
 # in the body string is already indented
 # But we clean the white space empty line afterward \o/
-def wrote_event_dispatchers(folder, t)
+def wrote_event_dispatchers(folder, t, m, callback_types)
   event_name = 'event'
+
+  matching_dispatchers = m.map do |md|
+    event, clock = t.find_event_class { |e| Regexp.new(md[:regex]).match?(e.name) }
+    raise "Matching callback #{md[:name]} does not match any event in model." unless event
+
+    args = {}
+    event.get_getter(event: '', arg_variables: args)
+    
+    dispatcher_args = []
+    dispatcher_args += [GeneratedArg.new('int64_t', '_timestamp')] if clock
+    dispatcher_args += args.fetch('outputs',[]).filter_map { |a| a if md[:args].include?(a.name) }
+    raise "Matching callback argument not present in model." unless dispatcher_args.length == md[:args].length
+
+    DispatcherInfo.new(md[:name], dispatcher_args, md[:name], Regexp.new(md[:regex]))
```

Comment:
If you do `Regexp.new(md[:regex]` multiple time, just store is as a regex in the map